frontend/lib/src/components/elements/Popover/Popover.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+
+import { useTheme } from "@emotion/react"
+import { ExpandMore, ExpandLess } from "@emotion-icons/material-outlined"
+
+import { hasLightBackgroundColor } from "@streamlit/lib/src/theme"
+import { StyledIcon } from "@streamlit/lib/src/components/shared/Icon"
+import { Block as BlockProto } from "@streamlit/lib/src/proto"
+import BaseButton, {
+  BaseButtonKind,
+  BaseButtonSize,
+} from "@streamlit/lib/src/components/shared/BaseButton"
+import StreamlitMarkdown from "@streamlit/lib/src/components/shared/StreamlitMarkdown"
+import IsSidebarContext from "@streamlit/lib/src/components/core/IsSidebarContext"
+
+import { Popover as UIPopover, TRIGGER_TYPE, PLACEMENT } from "baseui/popover"
+
+import { StyledPopoverButtonIcon } from "./styled-components"
+
+export interface PopoverProps {
+  element: BlockProto.Popover
+  empty: boolean
+}
+
+const Popover: React.FC<PopoverProps> = ({
+  element,
+  empty,
+  children,
+}): ReactElement => {
+  const [open, setOpen] = React.useState(false)
+  const isInSidebar = React.useContext(IsSidebarContext)
+  const theme = useTheme()
+
+  const lightBackground = hasLightBackgroundColor(theme)
+
+  return (
+    <div data-testid="stPopover">
+      <UIPopover
+        triggerType={TRIGGER_TYPE.click}
+        placement={PLACEMENT.bottomLeft}
+        content={() => children}
+        isOpen={open}
+        onClickOutside={() => setOpen(false)}
+        onEsc={() => setOpen(false)}
+        ignoreBoundary={isInSidebar}
+        overrides={{
+          Body: {
+            props: {
+              "data-testid": "stPopoverBody",
+            },
+            style: () => ({
+              marginRight: theme.spacing.lg,
+              marginBottom: theme.spacing.lg,
+              maxHeight: "70vh",
+              overflow: "auto",
+              maxWidth: "80vw",
+              minWidth: "20rem",
+
+              paddingRight: "calc(1em - 1px)", // 1px to account for border.
```

Comment:
It might be a good idea to put the border directives directly here to see where the `1px` comes from. Also, is it possible to use a variable to prevent issues in case the border will be changed in the future?

Reply:
You mean the css props that define the border width? they are directly underneath, but I moved them on top of the padding.